src/dumpulator/memory.py

```diff
@@ -104,28 +107,28 @@ def read(self, addr: int, size: int) -> bytearray:
     def write(self, addr: int, data: bytes) -> None:
         raise NotImplementedError()
 
+@dataclass
 class MemoryBasicInformation:
-    def __init__(self, base: int, allocation_base: int, allocation_protect: MemoryProtect):
-        self.base = base
-        self.allocation_base = allocation_base
-        self.allocation_protect = allocation_protect
-        self.region_size: int = PAGE_SIZE
-        self.state: MemoryState = None
-        self.protect: MemoryProtect = None
-        self.type: MemoryType = None
-        self.info: List[Any] = []
+    base: int
+    allocation_base: int
+    allocation_protect: MemoryProtect
+    region_size: int = PAGE_SIZE
+    state: Optional[MemoryState] = None
+    protect: Optional[MemoryProtect] = None
+    type: Optional[MemoryType] = None
+    info: List[Any] = field(default_factory=list)
 
     def __str__(self):
         return f"MemoryBasicInformation(base: {hex(self.base)}, allocation_base: {hex(self.allocation_base)}, region_size: {hex(self.region_size)}, state: {self.state}, protect: {self.protect}, type: {self.type})"
 
+@dataclass
 class MemoryManager:
-    def __init__(self, page_manager: PageManager, minimum=0x10000, maximum=0x7fffffff0000, granularity=0x10000):
-        self._page_manager = page_manager
-        self._minimum = minimum
-        self._maximum = maximum
-        self._granularity = granularity
-        self._regions: List[MemoryRegion] = []
-        self._committed: Dict[int, MemoryRegion] = {}
+    _page_manager: PageManager
+    _minimum: int = 0x000000010000
+    _maximum: int = 0x7fffffff0000

```

Comment:
Either this should be 64-bits wide or as it was before, this is confusing

Reply:
Fixed by #[6b7dc90](https://github.com/mrexodia/dumpulator/pull/47/commits/6b7dc9060c87a761a88e825e829a88ee845a79c0)